FashionConnect_ui/src/components/AddToCollectionsPopup/AddToCollectionsPopup.jsx

```diff
@@ -1,96 +1,111 @@
 import { Link } from "react-router-dom";
 import "./AddToCollectionsPopup.css";
-import React, {useState} from "react";
+import React, { useState, useEffect } from "react";
 
 
 
-export default function AddToCollectionsPopup({ collections, productID, onClose }){
+export default function AddToCollectionsPopup({ productID, userId, onClose }) {
 
-    const [selectedCollections, setSelectedCollections] = useState([]);
+  const [collections, setCollections] = useState([]);
+  useEffect(() => {
+    const fetchCollections = async () => {
+      try {
+        const user_collections = await fetch(`http://localhost:3000/collections/${userId}`);
+        const data = await user_collections.json();
+        setCollections(data);
+      } catch (error) {
+        throw error;
+      }
+    };
 
-    const handleCollectionChange = (event, collectionID) => {
-        if (event.target.checked) {
-          setSelectedCollections((prevSelected) => [...prevSelected, collectionID]);
-        } else {
-          setSelectedCollections((prevSelected) =>
-            prevSelected.filter((id) => id !== collectionID)
-          );
-        }
-      };
-    
-    const handleAddToCollection = () => {
+    fetchCollections();
+  }, [userId]);
+
+  const [selectedCollections, setSelectedCollections] = useState([]);
+
+  const handleCollectionChange = (event, collectionID) => {
+    if (event.target.checked) {
+      setSelectedCollections((prevSelected) => [...prevSelected, collectionID]);
+    } else {
+      setSelectedCollections((prevSelected) =>
+        prevSelected.filter((id) => id !== collectionID)
+      );
+    }
+  };
+
+  const handleAddToCollection = () => {
     // To Update the collections with the new ProductID
```

Comment:
when you have a comment that describes a function and what it's doing, it's best practice to put the comment before the function declaration

Reply:
Alright thanks!